### PR TITLE
Adds the Dorian Casino, a new space ruin.

### PR DESCRIPTION
**Pull request by warriorstar-orion**
## What Does This PR Do

This PR adds a new space ruin, the Dorian Casino.

_If you are looking for the most exclusive, luxurious casino in the galaxy, look no further. The Dorian Casino has everything you need to enjoy relaxing gameplay in an environment far away from the hustle and bustle of other gaming halls. With a top-shelf bar and unparalleled dining experience, the Dorian is **the** casino for those who demand exceptional entertainment experiences._

<details>
<summary>Technical Notes</summary>

This ruin is a powered ruin, and includes an SMES and small solar array. I didn't do any atmos stuff, typically because depressurization doesn't happen very often on ruins, but I may revisit it.

A new corpse spawner has been introduced which not only randomly chooses a species, but also allows for choosing random clothing elements. Right now only the shoes and uniform are randomly chosen but anyone can expand the implementation to support randomizing other slots. This allows for some dynamicism in the gambler corpses so they're not always the same exact spawns.

A half-built teleporter is present for explorers to add to the teleporter network. A safe in the security room contains a teleporter hub board.

</details>

<details>
<summary>Design Notes</summary>

The basic concept of the casino is to have an fancy, gaudy public area contrasted with small dingy maintenance area. The service tunnels are isolated to two cramped hallways completely out of sight of the elite clientele, the kitchen is hidden from the restaurant guests, the public facing walls are all valuable minerals in contrast to the regular walls in the staff areas, and so on.

I much prefer having real powernets rather than magical APCs/SMESes for realism and environmental reasons. If a ruin can always be explored in total brightness with functional airlocks I think that takes something away from it.

The lights in the gambling hall are a slightly dimmer blue. Casinos often use ambient lighting to encourage people to play more or stay longer at the tables, and dimmer lighting has been found to increase player's risk-taking.

As with all space ruins, the tables can have salvage spawn on them so investigating the entire ruin is worthwhile.

It is possible to retrieve silver from the walls. I can make a subclass for them where the silver isn't returned when deconstructing, if it's considered a problem.

There is a security camera console, but no cameras. I might just add them as decoration or actually hook them up to the console, dunno.

</details>

## Why It's Good For The Game

Having additional space ruins adds more variety to rounds and keeps the explorer role fresh.

## Images of changes

<details>
<summary>SDMM</summary>

StrongDMM-2024-06-04 14 01 23

</details>

<details>
<summary>Main Interiors</summary>

(note the high rate of salvage spawns are because this is the only ruin I'm spawning)

2024_06_04__14_05_40__Paradise Station 13
2024_06_04__14_06_16__Paradise Station 13
2024_06_04__14_06_35__Paradise Station 13

2024_06_04__14_05_02__Paradise Station 13
2024_06_04__14_05_22__Paradise Station 13
2024_06_04__14_05_32__Paradise Station 13

</details>

<details>
<summary>Samples of randomized gambler corpses</summary>

2024_06_04__14_25_38__Paradise Station 13

2024_06_04__14_25_46__Paradise Station 13

</details>

## Testing

Spawned in, ensured power network worked, walked around ruin, ensured all areas were present and not broken.

## Changelog

:cl:
add: A new space ruin, the Dorian Casino, has been added.
/:cl:

### CODE REVIEW

**Comment by Fordoxia** [comment]
Very nice, I like it!

One thing: Probably remove the floor tiles from under the windows here.
image

File: code/modules/awaymissions/mob_spawn.dm [hunk 1]
```diff
@@ -598,6 +598,53 @@
...
+		/datum/species/vulpkanin,
+		/datum/species/tajaran,
+		/datum/species/slime,
+		/datum/species/kidan,
+		/datum/species/drask,
+		/datum/species/grey,
+		/datum/species/diona,
+	)
+
+	. = ..()
```
Comment (Sirryan2002) [inline code/modules/awaymissions/mob_spawn.dm:640]:
```suggestion
	return ..()
```

**Review by Sirryan2002 — APPROVED** [review]

**Comment by warriorstar-orion** [comment]
> Very nice, I like it!
> 
> One thing: Probably remove the floor tiles from under the windows here, just have bare plating (you can't really see the floor well at all because of the grilles anyway). image

Done!

**Review by Burzah — APPROVED** [review]
Looks good!

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**